src/OSPSuite.Core/Domain/Descriptors/EntityDescriptor.cs

```diff
@@ -1,18 +1,45 @@
-﻿namespace OSPSuite.Core.Domain.Descriptors
+﻿using System;
+
+namespace OSPSuite.Core.Domain.Descriptors
 {
    public class EntityDescriptor
    {
-      public Tags Tags { get; set; }
-      public Tag Container { get; set; }
+      public Tags Tags { get; }
+
+      //Ensure that we only compute the parent descriptor tags when required
+      private readonly Lazy<EntityDescriptor> _lazyParentDescriptor;
+      private Tags _parentContainerTags;
 
-      public EntityDescriptor()
+      public EntityDescriptor(IEntity entity) : this(entity, createDescriptor)
       {
       }
 
-      public EntityDescriptor(IEntity entity)
+      public EntityDescriptor(IEntity entity, Func<IEntity, EntityDescriptor> createDescriptor)
       {
-         Tags = new Tags(entity.Tags){new Tag(entity.Name)};
-         Container = new Tag(entity.ParentContainer?.Name);
+         Tags = new Tags(entity.Tags) {new Tag(entity.Name)};
+         _lazyParentDescriptor = new Lazy<EntityDescriptor>(() => createDescriptor(entity.ParentContainer));
+      }
+
+      private static EntityDescriptor createDescriptor(IEntity entity) => entity == null ? null : new EntityDescriptor(entity);
+
+      public Tags ParentContainerTags
```

Comment:
For the caller, this should be transparent/
When ParentContainerTag is retrieved, we actually create the tag if not availalble
Then we also retrieve the hierarchy of tags